src/server/server.cpp

```diff
@@ -1,17 +1,53 @@
 #include "server.hpp"
+#include "reply.hpp"
+#include "text_data.hpp"
 
-Server::Server(int t_port): m_conn(t_port, new TextIO()), m_address(SERVER_ADDRESS){
-    m_conn.setup(m_address);
-    m_conn.bind_and_listen(m_address);
+Server::Server(int t_port)
+    : m_conn(t_port, new TextIO()), m_address(SERVER_ADDRESS) {
+  m_conn.setup(m_address);
+  m_conn.bind_and_listen(m_address);
+}
+
+void Server::main_loop() {
```

Comment:
When does this loop break?

Reply:
It will never break because the server always be running. If the server stops running all the clients will stop working.